OpenGL/Math/Matrix3.cs

```diff
@@ -98,6 +98,61 @@ public static Matrix3 Identity
             }
         }
 
+        public float this[int column, int row]
+        {
+            get
+            {
+                if (row == 0)
+                {
+                    if (column == 0) return row1.X;
+                    if (column == 1) return row1.Y;
+                    if (column == 2) return row1.Z;
+                }
+                if (row == 1)
+                {
+                    if (column == 0) return row2.X;
+                    if (column == 1) return row2.Y;
+                    if (column == 2) return row2.Z;
+                }
+                if (row == 2)
+                {
+                    if (column == 0) return row3.X;
+                    if (column == 1) return row3.Y;
+                    if (column == 2) return row3.Z;
+                }
+                if (column > 2 || column < 0 || row > 2 || row < 0) 
+                    throw new ArgumentOutOfRangeException("column or row is out of the matrix bounds.");
+                else return -1;
+            }
+            set
+            {
+                bool set = false;
+                if (row == 0)
+                {
+                    if (column == 0) row1.X = value; set = true;
```

Comment:
Once you have determined the row, why not just use `row1[column]`?  That would eliminate all of the `column` checks in this code.